### ISSUE TITLE
Regression: 2.6 sets human players to Weak AI in network host screen.

### ISSUE THREAD

**Issue by asvitkine** [triplea-game/triplea#10734]
### Reproduction Steps

1. Start a local game on Imperialism 1974 Board Game. Observe this screen:

![image](https://user-images.githubusercontent.com/17648/175455464-2c3c8e58-40e1-4ac4-93e7-edffc253e495.png)

2. Start a network game on the same map. Observe this screen:

![image](https://user-images.githubusercontent.com/17648/175455446-fad54a84-f041-4a69-9cf7-af90c8e7b139.png)

Notably, the XML for this map is:

```
		<player name="Blue" optional="true" canBeDisabled="true"/>
		<player name="Red" optional="true" canBeDisabled="true"/>
		<player name="Green" optional="true" canBeDisabled="true"/>
		<player name="Yellow" optional="true" canBeDisabled="true"/>
		<player name="Orange" optional="true" canBeDisabled="true"/>
		<player name="Cyan" optional="true" canBeDisabled="true"/>
		<player name="Purple" optional="true" canBeDisabled="true"/>
		<player name="Brown" optional="true" canBeDisabled="true"/>
		<player name="Neutrals" optional="true" defaultType="DoesNothing" isHidden="true"/>
```
So, the problems on the network screen are:
  1. WeakAI set as the non-Neutrals players, instead of Human.
  2. HardAI instead of "Does Nothing AI" set as the Neutrals player.

### Engine Version
The above screenshots are from 2.6 prerelease.

When using 2.5, the network game screen is as follows instead:
<img width="513" alt="Screen Shot 2022-06-23 at 11 20 27 PM" src="https://user-images.githubusercontent.com/17648/175455576-3cffdc7a-506e-4282-8b5e-2a12561af16c.png">

So the WeakAI change in 2.6 is incorrect, and both 2.5 and 2.6 are setting the `defaultType="DoesNothing"` to "Hard AI" incorrectly.

**Comment by asvitkine** [comment]
@RoiEXLab, can you look if the WeakAI change was a result of your recent refactoring?

Separately, perhaps some of the code between the two screens can be consolidated so the logic is the same?
I think one piece of the code is in `ServerSetupPanel.java`, where there's special logic based on player names (e.g. `playerName.startsWith("Neutral")`) and the other piece of logic is in `PlayerSelectionRow.java`, which uses `player.isDefaultTypeAi() and `player.isDefaultTypeDoesNothing()`.

**Comment by RoiEXLab** [comment]
@asvitkine I can look into this later this day but I doubt that the map setting being ignored has something to do with the refactoring. I haven't seen anything related parsing related there, but at the same time i could also just have missed it

**Comment by RoiEXLab** [comment]
@asvitkine I have trouble reproducing your issue. Mainly because I don't appear to have the map you described locally and I can't open the download maps screen because the servers are not reachable (is this intended?). Commit a1a2188e0648d172136218b7b8cd89fbcfc6664a is the one before my "changes to the AI" if you want to try that out. I'm still convinced #10709 does not introduce this issue, but I can't guarantee I have nothing to do with this issue because I touched the relevant classes several times before.

**Comment by asvitkine** [comment]
@RoiEXLab downloads broken in 2.6 is a known Issue that I think @DanVanAtta needs to fix.

The workaround is to download using 2.5.

**Comment by asvitkine** [comment]
I tried a network game with https://github.com/triplea-game/triplea/releases/tag/2.6.955 which is right before your PR and I confirm the players get set to Human when doing "Host Networked Game" game. So it seems your change is responsible for this. Re-assigning to you.

**Comment by RoiEXLab** [comment]
Thanks for having a look. Still not entirely sure why this happens, but I'll see what I can do. Might take me some time though.

**Comment by DanVanAtta** [comment]
The prerelease server is still running with 2.6
There is no prod 2.6 server yet. The vanilla 2.6 configuration is to use 'prod' and that would be contacting a 2.5 server. To resolve this, update 'settings > testing' to use the prerelease server. The plan is to remove the prerelease option and deploy a 2.6 prod server which would allow 2.6 clients to work out of the box.

**Comment by RoiEXLab** [comment]
@asvitkine I have a fix for the regression. It's really dumb. Would you mind creating a separate issue for the map settings discrepancy? So I can tag this issue to be closed once the PR is merged?

**Comment by asvitkine** [comment]
> @asvitkine I have a fix for the regression. It's really dumb. Would you mind creating a separate issue for the map settings discrepancy? So I can tag this issue to be closed once the PR is merged?

Sure, I've changed the title. I can file a new issue once the fix is in, so I can repro the exact behavior.

**Comment by asvitkine** [comment]
> The prerelease server is still running with 2.6 There is no prod 2.6 server yet. The vanilla 2.6 configuration is to use 'prod' and that would be contacting a 2.5 server. To resolve this, update 'settings > testing' to use the prerelease server. The plan is to remove the prerelease option and deploy a 2.6 prod server which would allow 2.6 clients to work out of the box.

1. Why do they need separate servers?
2. If there's a good reason for 1, why not make 2.6 use the prerelease server by default?

**Comment by asvitkine** [comment]
@DanVanAtta 

Also, I don't see a download server option under testing. Do you mean the lobby server?
<img width="1132" alt="Screen Shot 2022-06-25 at 4 50 47 PM" src="https://user-images.githubusercontent.com/17648/175790165-4b3ed8e3-0244-435a-ab84-d2411354b480.png">

Setting that, I still get an error:
<img width="1552" alt="Screen Shot 2022-06-25 at 4 51 38 PM" src="https://user-images.githubusercontent.com/17648/175790170-67d84c58-8cd1-43dc-92f7-582a3b3c98e6.png">
